aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -1,51 +1,128 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::aggregator_extension::AggregatorID;
+use crate::{
+    aggregator_extension::AggregatorID,
+    delta_change_set::{serialize, DeltaOp},
+    module::AGGREGATOR_MODULE,
+};
+use aptos_types::{
+    state_store::{
+        state_key::StateKey,
+        state_value::{StateValue, StateValueMetadataKind},
+    },
+    write_set::WriteOp,
+};
+use move_binary_format::errors::Location;
+use move_core_types::vm_status::{StatusCode, VMStatus};
 
-/// Defines different ways a value of an aggregator can be resolved in
-/// `AggregatorResolver`. The implementation of the trait can use custom
-///  logic for different reading modes.
+/// Defines different ways `AggregatorResolver` can be used to read its value
+/// from the state.
 pub enum AggregatorReadMode {
     /// The returned value is guaranteed to be correct.
     Precise,
```

Comment:
I think we should adjust the descriptions here, or even the names, the way it's implemented now:
at the end of the day, even "Speculative" is guaranteed to be correct, as long as previous transactions have been committed (or in sequential execution).
Precise is precise in sequential (but so is Speculative), but in parallel, it isn't guaranteed to be precise either. Yes, it's validated, and in that sense yes, but otherwise, it is actually a speculative value.

In the extension we used Aggregated and LastCommitted. I feel those are fine and quite descriptive. We could also distinguish by the fact that Aggregated is validated, but maybe that can be a comment. The description is simply that Aggregated takes any available deltas from previous transactions into account (when executing in a speculative context), while LastCommitted does not. Also important to mention that LastCommitted isn't validated and should be used for well-thought out specific purposes only where validation is provided differently (e.g. branch prediction). And for sequential execution (nothing is speculative), they are equivalent.

Reply:
I've updated the names and descriptions on aggregators_v2 branch, didn't realize this was on main. I'm fine with leaving it like this on main, and having it changed on the branch only

Reply:
Oh I was also confused, didn't notice only the comment was changed. Changing when we merge the aggregatorV2 branch sounds good